proto/api/v1/discovered_cluster_service.proto

```diff
@@ -0,0 +1,135 @@
+syntax = "proto3";
+
+package v1;
+
+import "api/v1/pagination.proto";
+import weak "google/api/annotations.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "v1";
+option java_package = "io.stackrox.proto.api.v1";
+
+// DiscoveredCluster is the service representation of discovered clusters.
+//
+// A discovered cluster originates from a cloud source integration. If
+// an equivalent internal cluster exists, the discovered cluster is considered
+// secured. If equivalent secured cluster exists, the status of the discovered
+// cluster is either unsecured or unknown, depending on the provider details.
+message DiscoveredCluster {
+  // UUIDv5 generated deterministically from the tuple (metadata.id, metadata.type, source.id).
+  string id = 1;
+
+  message Metadata {
+    // Represents a unique ID under which the cluster is registered with the
+    // cloud provider. Matches storage.ClusterMetadata.id for secured clusters.
+    string id = 1;
+    // Represents the name under which the cluster is registered with the
+    // cloud provider. Matches storage.ClusterMetadata.name for secured clusters.
+    string name = 2;
+    enum Type {
+      UNSPECIFIED = 0;
+      AKS = 1;
+      ARO = 2;
+      EKS = 3;
+      GKE = 4;
+      OCP = 5;
+      OSD = 6;
+      ROSA = 7;
+    }
+    // The cluster type. Matches storage.ClusterMetadata.type for secured clusters.
+    Type type = 3;
+
+    enum ProviderType {
+      PROVIDER_TYPE_UNSPECIFIED = 0;
+      PROVIDER_TYPE_AWS = 1;
+      PROVIDER_TYPE_GCP = 2;
+      PROVIDER_TYPE_AZURE = 3;
+    }
+    // The cloud provider type.
+    ProviderType provider_type = 4;
+
+    // The region as reported by the cloud provider.
+    string region = 5;
+    // Timestamp at which the cluster was first discovered by the cloud source.
+    google.protobuf.Timestamp first_discovered_at = 6;
+  }
+  Metadata metadata = 2;
+
+  enum Status {
+    // The status of the cluster is unknown. May occur if a secured cluster
+    // is missing the metadata for a possible match.
+    STATUS_UNSPECIFIED = 0;
+    // The discovered cluster was matched with a secured cluster.
+    STATUS_SECURED = 1;
+    // The discovered cluster was not matched with a secured cluster.
+    STATUS_UNSECURED = 2;
+  }
+  Status status = 3;
+
+  message CloudSource {
+    string id = 1;
+    string name = 2;
```

Comment:
This is probably something we should talk in the next sync together with the UI team: whether they _really_ require this, or it is better to have them fetch the name by the given ID.

Reply:
I left fetching the name out of the service for now. Indeed let's align with the UI team.